esp32-mock-generator/src/main.cpp

```
@@ -0,0 +1,319 @@
+  JsonArray harmonicsVArray = doc["harmonics_v"].to<JsonArray>();
+  JsonArray harmonicsIArray = doc["harmonics_i"].to<JsonArray>();
+
+  for (int i = 0; i < 8; i++) {
+    harmonicsVArray.add(round(harmonicsV[i] * 100) / 100.0);
+    harmonicsIArray.add(round(harmonicsI[i] * 100) / 100.0);
+  }
+
+  // -------------------------------------------------------------------------
+  // 5. Publish to MQTT
+  // -------------------------------------------------------------------------
+  char buffer[512];
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)